### DISCUSSION TITLE
Numerical instability in simulating solid particle ejection

### DISCUSSION THREAD

**Discussion by ahsadeghi** [firemodels/fds#12654] [Q&A]
Hello,
Following the below discussions on FDS google group 
https://groups.google.com/g/fds-smv/c/iYC4bYUagkk
https://groups.google.com/g/fds-smv/c/iT7dmOoJuCQ/m/CxZIStqyAAAJ

I still have problems with particle ejection. Thanks to Kevin and dr_jfloyd for the comments, but the solutions did not work for my case. Because I cannot change the particle properties, particle diameter, mass flow rate, and vent area, as they are from experiments and I need them for validating the simulations. I have tried different ways such as decreasing DRAG_COEFFICIENT, defining VEL_PART, changing MAX_PRESSURE_ITERATIONS, including SUSPEND_PRESSURE_ITERATIONS=T, but they didn't help. I wonder if you would please let me know if you have any solutions. Thank you for your consideration. Please find attached the input file. It is noteworthy that thermal conductivity is high to have the lumped capacitance assumption.

[Model_small.txt](https://github.com/firemodels/fds/files/14669245/Model_small.txt)

Regards,
Hosein

**Comment by drjfloyd** [comment]
What version of FDS are you using?
What is it exactly you are trying to do with the particles? What behavior are you expecting to see and what behavior are you seeing?

**Reply by ahsadeghi** [reply]
The version is FDS-6.8.0-1238-g64f51df
I am trying to simulate particle ejection from a 18650-type battery. I expect the HRR, particle mass flow, the total particle mass of 4 g and the  particle size shown in the figure below. 

![image](https://github.com/firemodels/fds/assets/51397945/9c4f8559-554d-4312-bb72-860cf22cc191)

the majority of these data are from the following paper:
https://www.sciencedirect.com/science/article/pii/S0378775323007334

Regards,
Hosein

**Comment by drjfloyd** [comment]
How can you know you are not getting that mass flux and size distribution? Your input file has no outputs measuring anything about particles. Your input file has no outputs specified at all.

&MATL ID='Particle_mat', 
      DENSITY=1.0, 
	  CONDUCTIVITY=1e10, 
	  SPECIFIC_HEAT=0.8,
      EMISSIVITY= 0.9/

The DENSITY is not realistic. This is the density of air. A solid particle is going to have a density much larger than this. The paper gives 2000 kg/m3, If your goal is to replicate the paper, you should be using the values in the paper unless you have a good argument as to why those values are not correct.

You don't need such a high CONDUCTIVITY. Non-physical values of parameters can lead to solver issues.  FDS sets the wall node size using the equation in 8.3.8 of the guide.  You only need CONDUCTIVITY ~0.1 W/m/K for FDS to have just one wall node.

I would suggest you start with a much simpler input file. Just inject monodisperse particles from a surface and figure out how to measure the mass of particles being injected. Then maybe add a fire to that surface as well. Then maybe add a CNF distribution. Then maybe add time depdendent ramps for the the fire and particle flux.

&VENT ID='Burner', SURF_ID='Burner', XB=0.032,0.042,0.03,0.04,0.066,0.066/ 

You have a 1 cm x 1 cm surface for your particle flux. While this may be roughly the size of an 18650 cell, unless the top of the cell blew off during the experiment, the particles and gas flow are coming from a smaller area.  Using a larger area will reduce the injection velocity.

You are using a CNF_RAMP. CNF stands for cummulative number fraction. The plot in the paper you are basing the ramp off of is a mass fraction plot. The two are not the same.  Say you have 10 um particles and 100 um particles with a 50 % mass fraction in each.  A 10 um particle has 0.1 % of the mass of a 100 um particle. In number fraction this means you are not 50 % 10 um and 50 % 100 um but 99.9 % 10 um and 0.1 % 100 um.

Why are you setting  N_STRATA=1?  

&REAC ID='Reaction1',
      FUEL='VENTED GASES'
	  FORMULA='C1H4'/

Why 100 % methane your fuel (METHANE is also a predefined species and you should use that rather than the formula)? The paper states: "the composition of the gas mixture generated by the NCM LIB with 50%SOC is 20.28% H2, 38.65% CO, 30.16% CO2, 4.66% CH4, 5.9% C2H4 and 0.35% C2H6"

**Comment by ahsadeghi** [comment]
Thank you for your comment. I have done a simpler case before this case, please see the following input file, this case has a bigger surface area and the boundaries are not open to measure the total mass of the particles (I know it can affect the HRR but here I was verifying the particle part)

[particle_test.txt](https://github.com/firemodels/fds/files/14670481/particle_test.txt)

sorry the correct density is 2000 kg/m3, I was testing with 1 kg/m3 to see if the problem is with the weight of the particles. 
I created the CNF_RAMP ramp in the input file based on the figure, the data in the figure were not used in the input file directly. Then I designed the ramp for the mass flow rate to outcome a total mass of 4 g. There is a device in the newly attached input file to measure it. 

I had N_STRATA=1 because of having CNF_RAMP, similar to the example in section 15.4.4. However, removing it does not help. 

I had CH4 for simplicity, I included the original mixture but there is still the numerical instability. Please see the input file below. 

[Model_small.txt](https://github.com/firemodels/fds/files/14670740/Model_small.txt)

I have been working on this problem for a couple of weeks but only decreasing the mass flow rate of the particles to 0.0000133 (in Model_small.txt input file) helped, which is not desired. This amount seems like not having particles. Otherwise, changing other parameters only changed the instability time around 0.1-0.2 seconds. 


Regards,
Hosein

**Comment by drjfloyd** [comment]
The cell vent is 9 mm2 and you are modeling it at 100 mm2. The particles are going to be coming out at a decent velocity from the real vent. I added a VEL_PART, turned on PARTICLE_CFL, and injected more particles (NPPC=10).  I think your problem was with a small number of particles being injected at near zero velocity that they were collecting on the top and eventually the resulting drag made things go unstable.

[Model_small.fds.txt](https://github.com/firemodels/fds/files/14672175/Model_small.fds.txt)

**Comment by rmcdermo** [comment]
Is there any reason why the particles would not be exiting the VENT at the same velocity as the fuel gas?  To me, that would mean ramping the part velocity as well.  The fuel velocity is (HRRPUA/HOC)/RHO_FUEL, which should peak around 0.3 m/s = (10,000 kW/m2 / 50,000 kJ/kg) / (0.667 kg/m3).  But maybe I'm missing something.

**Comment by drjfloyd** [comment]
In the real world there is a very tiny vent hole on the battery top that fails first. It is much smaller than the overall battery diameter. In the FDS input the VENT is placed over the entire battery top which means the battery gases are at a much lower velocity than if they just came from the vent hole.

Also at early times the RAMP for the flows is near zero.

**Comment by rmcdermo** [comment]
OK, but in the real case, with a much smaller orifice, still the gas and particles are at roughly the same velocity, I would think.  I am just saying that we should probably keep that relative velocity in the model if we want to avoid spurious drag.  Yes, accurate account of the jet momentum is lost once you smear it out to 10 times the area.

**Comment by ahsadeghi** [comment]
Thank you @rmcdermo and @drjfloyd. The simulation is still running with the changes you made @drjfloyd. Yes, you are right, I should fix the vent area.

**Comment by maria-justo-alonso** [comment]
Hello,
This is a kind of follow up question
I am beginner in FDS.I am simulating the ventilation in a battery room.  I would like to set up the thermal runaway of a battery module where the cells of a module enter in TR when the neighboring wall of the cell has a temperature higher than 380C. I know that it is possible to set a timer. But is possible to set a minimum temperature for starting the fire?
Then as there are two parts of the room very differentiated, one with a combustion and the other where there is only the mixing of particles due to ventilation I guess it is better to have different mesh's sizes. Would you have any advice how to do that? 
Thanks in advance! Maria

**Comment by mcgratta** [comment]
You can use `IGNITION_TEMPERATURE` and a `HRRPUA` ramp on a `SURF` line to define the boundary conditions of your battery pack. When the surface temperature reaches the specified ignition temperature, the object will burn with the given `HRRPUA`.

**Reply by maria-justo-alonso** [reply]
So would you make every cell in the battery pack as something like this: 
&OBST XB= 0.05, 0.15, 1.8, 1.9, 0, 0.15, COLOR='RED', SURF_ID='fire1' /
&SURF ID='fire1',IGNITION_TEMPERATURE =380, HRRPUA=250 /
And then one mesh containing the battery pack and another for the rest of the room?

**Reply by maria-justo-alonso** [reply]
Hi again. I have managed the delayed ignition in the cell, however, I need some help with the other cells in teh pack. I have defined them as objects as it follows: 
&MATL ID='STEEL',
      FYI='Drysdale, Intro to Fire Dynamics - ATF NIST Multi-Floor Validation',
      SPECIFIC_HEAT=0.46,
      CONDUCTIVITY=45.8,
      DENSITY=7850.0,
      EMISSIVITY=0.95/

&SURF ID='fire1', TMP_INNER=40.0,
      MATL_ID(1,1)='STEEL',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS(1)=6.0E-4, IGNITION_TEMPERATURE =300, HRRPUA=0.380513346 /  

&OBST XB= 0.4475, 0.8000, 2.1742, 2.2000, 1.3998, 1.5000, COLOR='BLUE',THICKEN=.TRUE., SURF_ID='fire1' /

However, it seems that there is no heat transfer to these objects. They did not increase the temperature when the neighboring cell is burning and even when the room air has temperature over 300 close to the burning cell, the other cells do not change the temperature at all. Am I defining the object wrong and should define it in a different manner? Thanks!

**Reply by mcgratta** [reply]
Try without `TMP_INNER` and make sure you have an external heat source that will obviously change the temperature. If that does not work, create a simple test case and start an issue.

**Reply by maria-justo-alonso** [reply]
I have removed it as recommended, however, the temperature in the objects is not changing. When I plot the slices and I ask to display only the contours of the objects, I have no color information within the objects outlines, the slices are empty within the object.

**Reply by drjfloyd** [reply]
An SLCF of TEMPERATURE when you have 1D heat tranfers is a gas phase only output. The volume covered by the OBST is a solid; there is no gas there. See the table of solid phase outputs in the User's Guide.

**Reply by maria-justo-alonso** [reply]
In my case I have a TR in the battery . I have one simple case with one supply and 1 exhaust and I want the ventilation to shut down when the temperature in a sensor reaches teh 68 C. I have programmed it like this: # Smoke Detector Device for Turning Off Ventilation at 68°C
&DEVC ID='temp_68', XYZ=1.835, 2.345, 2.4, QUANTITY='TEMPERATURE', SETPOINT=68., INITIAL_STATE=.TRUE. /

# Control to Deactivate Ventilation at 68°C or Above
&CTRL ID='deactivate_ventilation', FUNCTION_TYPE='ANY', INPUT_ID='temp_68', LATCH=.TRUE. /

# Supply Vent controlled by Deactivation Control
&VENT XB=0,0,1.5,2.3,0.2,0.6, SURF_ID='supply', CTRL_ID='deactivate_ventilation', ID='supply_vent' /
&SURF ID='supply', VOLUME_FLOW=-0.625, COLOR='BLACK' /

# Exhaust Vent controlled by Deactivation Control
&VENT XB=3.67,3.67,3.5,4,2.1,2.3, SURF_ID='exhaust', CTRL_ID='deactivate_ventilation', ID='exhaust_vent' /
&SURF ID='exhaust', VOLUME_FLOW=0.625, COLOR='BLACK' /
However, when I run the simulation even if the temperatures in the room are not very high, the flow in the supply is reversed becoming an exhaust. Do you have any idea why this may be happening? Do you have any recommendation how to do it otherwise?
[1c1e.txt](https://github.com/user-attachments/files/17926049/1c1e.txt)

**Reply by drjfloyd** [reply]
Please provide a complete simple input file that demonstrates your issue.  A fixed VOLUME_FLOW should not change direction. Most likely there is some input error elsewhere or there is some misinterpretation of the output. Without an input file we cannot diagnose further

**Reply by maria-justo-alonso** [reply]
It is here. Thanks!

**Reply by drjfloyd** [reply]
See 20.10.9 Volume Flow in the User's Guide for details on measuring the volume flow at a solid boundary.

**Reply by maria-justo-alonso** [reply]
Sorry, I dont understand this reply. Do you mean that in this chapter of the guide it says that the if the flow is in the positive direction and therefore both results are supposed to be positive as they go in the same direction?

**Reply by drjfloyd** [reply]
No. If you want to measure flow at the wall, use the NORMAL VELOCITY output format described in that chapter.

**Reply by maria-justo-alonso** [reply]
Hello, it seems taht I am still doing several things wrong. Hope I can get a little bit more help!I am trying to simulate the thermal runaway in once cell in a battery room.  I define that the fire should tsart in a cell in the second-row number 15 like this: 
&OBST XB=0.4360, 0.4520, 1.8000, 2.1520, 0.2050, 0.3050, COLOR='RED', THICKEN=.TRUE., SURF_ID='Burner'/

All the other cells should not sturt burning until the temperature is over 300 what I define like this: 
&OBST XB=0.1000, 0.1160, 1.8000, 2.1520, 0.3850, 0.4850, COLOR='CYAN', THICKEN=.TRUE., SURF_ID='fire1' /
&SURF ID='fire1', 
      MATL_ID(1,1)='STEEL',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS(1)=0.02, IGNITION_TEMPERATURE =300, PART_ID='Particle',
      NPPC = 10,
      VEL_PART=-10,
      PARTICLE_MASS_FLUX=1.33,
      RAMP_PART= 'Particle_ramp_mass_flux'
      HRRPUA= 7656.25,
      RAMP_Q='Burner_RAMP_Q', THICKNESS(1)=0.3/  

However, when studying the results in my ctrl.csv and dev.csv files, it seems that the fire is not starting in row 2 , cell 15, but it seems it starts in row 1 cell 1. Why is that?. What am I doing wrong?

Regarding my ventilation, I want that the ventilation is running constantly until I get ahigh temperature in the center of the room and at this point, the ventilation should stop both in supply and both exhausts.I define this in the simulation as follows: 
# Smoke Detector Device for Turning Off Ventilation at 68°C
&DEVC ID='temp_68', XYZ=1.835, 2.345, 2.4, QUANTITY='TEMPERATURE', SETPOINT=68., INITIAL_STATE=.TRUE. /

# Control to Deactivate Ventilation at 68°C or Above
&CTRL ID='deactivate_ventilation', FUNCTION_TYPE='ANY', INPUT_ID='temp_68', LATCH=.TRUE. /

# Supply Vent controlled by Deactivation Control
&VENT XB=0,0,1.5,2.3,0.2,0.6, SURF_ID='supply', CTRL_ID='deactivate_ventilation', ID='supply_vent' /
&SURF ID='supply', VOLUME_FLOW=0.625, COLOR='BLACK' /

# Exhaust Vent controlled by Deactivation Control
&VENT XB=3.67,3.67,3.5,4,2.1,2.3, SURF_ID='exhaust', CTRL_ID='deactivate_ventilation', ID='exhaust_vent' /
&SURF ID='exhaust', VOLUME_FLOW=-0.625, COLOR='BLACK' /

# Second Exhaust Vent controlled by Deactivation Control
&VENT XB=1.835,1.835,4.69,4.69,0,0.2, SURF_ID='exhaust_2', CTRL_ID='deactivate_ventilation', ID='exhaust_vent_2' /
&SURF ID='exhaust_2', VOLUME_FLOW=-0.625, COLOR='BLACK' /


# Output the supply and exhaust airflow rates
&DEVC ID='supply volume flow', XB=0,0,1.5,2.3,0.2,0.6, QUANTITY='NORMAL VELOCITY', SURF_ID='supply',SPATIAL_STATISTIC='SURFACE INTEGRAL' /
&DEVC ID='exhaust volume flow', XB=3.67,3.67,3.5,4,2.1,2.3,QUANTITY='NORMAL VELOCITY', SURF_ID='exhaust',SPATIAL_STATISTIC='SURFACE INTEGRAL' /
&DEVC ID='exhaust_2 volume flow', XB=3.67,3.67,3.5,4,2.1,2.3,QUANTITY='NORMAL VELOCITY', SURF_ID='exhaust_2',SPATIAL_STATISTIC='SURFACE INTEGRAL' /

However, when examining the results the ventilation in supply and exhaust is as defined in the first exhaust. But zero in the second exhaust. Any advice what I can be doing wrong here?
Another thing it is surprising is that although in the first row I get temperatures close to 600, at row 4 and upwards, the temperatue in teh gaps is not changing at all being constant 20, while at teh sensor in the top of teh room the temperature rises. Am I defining the sensors in between the cell rows wrongly?

Finally I am measuring the pressure in the exhaust vent and I have defined as this even though it is the in the boundary:
&DEVC XB=3.67,3.67,3.5,4,2.1,2.3, QUANTITY='PRESSURE', ID='exhaust pressure' /
Is it ok to measure like this or should I use  some kind of surface integral as I use in the volume flow?
[sample1c2e_devc.csv](https://github.com/user-attachments/files/18035093/sample1c2e_devc.csv)
[sample1c2e_devc_ctrl_log.csv](https://github.com/user-attachments/files/18035094/sample1c2e_devc_ctrl_log.csv)

**Reply by drjfloyd** [reply]
RE fire not starting:  Have you looked to see if your OBST is reaching 300 C?  If not, it isnt going to ignite.  It is going to take a lot of heat to get the surface of 2 cm thick piece of steel to reach 300 C.

You don't need the CTRL function. You can just use the TEMPERATURE DEVC for the VENT (i.e., DEVC_ID instead of CTRL_ID).

This is what is in the input file you attached:

# Second Exhaust Vent
#&VENT XB=1.835,1.835,4.69,4.69,0,0.2, SURF_ID='exhaust_2', CTRL_ID='vent_off' /
#&SURF ID='exhaust_2', VOLUME_FLOW=0.625,FLOW_THRU=.TRUE., COLOR='BLACK' /

This VENT has been commented out.

**Reply by maria-justo-alonso** [reply]
Sorry, I made a mistake in the file I had uploaded. Now the correct one is there. The vent was not commented. 

Regarding the temperature, you can see that the temperature next to the obst in the first row is 600( this one is igniting but I don't know why), while the temperature in the second row defined with fire that should start after 10 seconds to ignite is still at 20.
 
Then regarding you comment about the CTRL_ID., how would you use it? I want that the ventilation shut down when the temperature in my sensor is over 68, that is why I did the CTRL_ID.
thanks!

**Reply by drjfloyd** [reply]
&DEVC ID='temp_68', XYZ=1.835, 2.345, 2.4, QUANTITY='TEMPERATURE', SETPOINT=68., INITIAL_STATE=.TRUE. /

This will output a logical state that starts at T and goes to F when the temperature exceeds 68 C.  You can add LATCH  to the DEVC so it only changes state once., then you just use the DEVC for your VENT. You don't need a CTRL, i.e.:

&VENT XB=1.835,1.835,4.69,4.69,0,0.2, SURF_ID='exhaust_2', DEVC_ID='temp_68'/

Nothing wrong with having the CTRL, it is just extra work that you don't need to do.

However, take close look at your second exhaust VENT.  What is the area of this VENT with the XB as you have it written?

IGNITION_TEMPERATURE is not the gas temperature, it is the solid surface temperature.  TEMPERATURE gives the gas temperature. You need to look at the WALL TEMPERATURE. That must reach 300 C for the OBST to ignite.

**Reply by maria-justo-alonso** [reply]
you are right! The exhaust _2 was wrong and now it makes the flow as I want:)
REgarding the ignition, I understand you point that the ignition happens when teh wall temperatrue is over 300 for the surfaces called fire. However, for the surface called burner, I understand it should be automatically ( with 10 seconds delay ) as teh surface is defined as follows: &SURF ID='Burner',
      COLOR='RED',
      PART_ID='Particle',
      NPPC = 10,
      VEL_PART=-10,
      PARTICLE_MASS_FLUX=1.33
      RAMP_PART= 'Particle_ramp_mass_flux'
      HRRPUA= 7656.25,
      RAMP_Q='Burner_RAMP_Q', THICKNESS(1)=0.3/. Therefore, for me is strange that &OBST XB=0.1,   0.116,  1.8000,  2.1520,  0.0250,  0.1250, COLOR='GRAY', THICKEN=.TRUE., SURF_ID='fire1' /
is getting to an air  temperatrue nearby around 600 , while the burner is not yet starting. I also understand that you are recommending to change &DEVC ID='temp_obst_2_1', XYZ=0.108, 1.9760, 0.31, QUANTITY='TEMPERATURE', SETPOINT=300 / to wall temperature?

**Reply by drjfloyd** [reply]
In your input file, the 'Burner' surface is only being applied to one OBST.  All of your other OBST have the 'fire1' surface which requires a surface temperature of 300 C for a wall cell to start burning.

This isn't doing anything in your simulation other than recording the gas temperature at the XYZ in the devc.csv output file. Yes you have set a SETPOINT, but no other input in your fds file uses the output of this device to do anything. Your 'fire1' surface uses the wall temperature for each cell in the OBST to decide if that wall cell should ignite.

 &DEVC ID='temp_obst_2_1', XYZ=0.108, 1.9760, 0.31, QUANTITY='TEMPERATURE', SETPOINT=300 / 

I suggest you make a very simple input file so that you can test your object igniton.

**Reply by maria-justo-alonso** [reply]
thanks a lot for your patience! 
I didnt use the &DEVC ID='temp_obst_1_1', XYZ=0.108, 2.1520, 0.1250, QUANTITY='WALL TEMPERATURE', IOR=2, SETPOINT=300 / because I was getting this error messsage ERROR(427): DEVC temp_obst_1_1 requires repositioning.

ERROR: FDS was improperly set-up - FDS stopped (CHID: sample1c2e)
And despite changing the coordinates many times, I did not mange to get the positioning, so I moved to the air temperature as an incorrect proxi. 
However, I dont understand why if during the 10 first seconds when my single cell 'Burner' is not burning, I get a rise in the air temperature near the  obst_1_1 although the heat should only start when teh burner starts to burn after 10 seconds according to the Burner_RAMP_Q. Also after the 10 seconds I expect the 'Burner to start giving away heat ( and thus warming the air around) but I see no change in it.

**Reply by drjfloyd** [reply]
Please carefully review your inputs:

&RAMP ID='Burner_RAMP_Q', T=5.0, F=0.0/
&RAMP ID='Burner_RAMP_Q', T=6.814195, F=0.010001/

This is not a 10 second delay. T=5 is a  5 s delay

&MESH XB=0, 3.67, 0, 4.69, 0, 2.4, IJK= 60, 80, 60 /,MULT_ID='M'/

3.67/60 = 0.061 m x resolution
4.68/80 = 0.058 m y resolution

&OBST XB=0.4360, 0.4520, 1.8000, 2.1520, 0.2050, 0.3050, COLOR='RED', THICKEN=.TRUE., SURF_ID='Burner'/

The closet grid boundaries to this OBST are 0.4282 m and 0.4893 m

Look at the OBST after this one:

&OBST XB=0.4600, 0.4760, 1.8000, 2.1520, 0.2050, 0.3050, COLOR='YELLOW', THICKEN=.TRUE., SURF_ID='fire1' /

It's closet grid boundaries are 0.4282 m and 0.4893 m. The same as the previous OBST. What does the User's Guide say happens when there are overlapping OBST?

Again I highly suggset you start with a much simpler input file where your opportunities for errors are less. Test one thing at a time. Once you are confident that you know how your inputs work, then add complexity.

**Reply by maria-justo-alonso** [reply]
Now I understand!! When I change my mesh, suddenly it works as it should!
thanks for all these great advices! I will rebuild the case stepwise as you advice!

**Comment by mcgratta** [comment]
I do not suggest you mix meshes at first. Try something simple to see if the concept works. Then you might explore grid refinement. But I warn you that putting one mesh inside another is a fragile construct and generally avoid it.

**Reply by maria-justo-alonso** [reply]
Ok, I will try that then!Thanks a lot!

**Comment by ahsadeghi** [comment]
Dear @drjfloyd 
Previously you mentioned here that CONDUCTIVITY ~0.1 W/m/K is enough for FDS to have just one wall node (lumped capacitance assumption), i.e., Bi=hL/k <0.1. Would you please explain how it is done in FDS? Do you have an approximation for h?

Regards,
Hosein

**Reply by drjfloyd** [reply]
8.3.8 Solid Phase Numerical Gridding Issues
